debian/ubuntu-advantage-tools.postinst

```diff
@@ -217,7 +217,7 @@ EOF
         cat > "${apt_pref_file}" <<EOF
 # Written by ubuntu-advantage-tools
 Package: *
-Pin: release o=${apt_origin}, n=${release}
+Pin: release o=${apt_origin}
```

Comment:
Ack. Undertstood that we want this config to be release-agnostic, because do-release-upgrade when calculating package diffs to upgrade will incorrectly preserve this preferences.d file without updating the n=${release} to the target upgraded release name which then leaves this upgrade path exposed to trying to install any new pkg from Bionic.

Dropping the `n=${release}` filter will allow the never pin to be honored across release upgrade paths

ESM_INFRA_APT_PREF_FILE is a release-agnostic filename `/etc/apt/preferences.d/ubuntu-esm-apps` so dropping the release-specific filter makes sense to me. Our postinst will also double check upon bionic package install if no configuration APT source yet exists for esm-apps